app/DataStructure.js

```diff
@@ -0,0 +1,71 @@
+import { ShowButton } from "./ShowButton";
+
+export class DataStructure {
+  constructor() {
+    this.usersDataURL = "http://localhost:3000/users";
+    this.companiesDataURL = "http://localhost:3000/companies";
+    this.fetchData();
+    this.companiesData = {};
+    this.usersData = {};
+  }
+  fetchData = async () => {
+    const usersResponse = await fetch(this.usersDataURL);
+    this.usersData = await usersResponse.json();
+    const companiesResponse = await fetch(this.companiesDataURL);
+    this.companiesData = await companiesResponse.json();
+    this.produceData();
+  };
+
+  produceData = () => {
+    this.sortedDataArray = [];
```

Comment:
`sortedDataArray` is still a poor name for object property. Pleas try something more descriptive. To be able to find a good name answer the following question to yourself "What is this array containing?"

If the array contain fruits you name the array `fruits`
If the array contains all the animals that are mammals, but onlt those containing horns, you name the array `mamalsWithHorns`

Reply:
changed as suggested